torch/_custom_op.py

```diff
@@ -786,3 +790,33 @@ def parse_return(annotation, error_fn):
 
 
 SUPPORTED_PARAM_TYPES = get_supported_param_types()
+
+
+def report_error_callback(custom_op: typing.Any, key: str) -> None:
+    if key == "Undefined":
+        raise NotImplementedError(
+            f"{custom_op}: There were no Tensor inputs to this operator "
+            f"(e.g. you passed an empty list of Tensors). If your operator is a "
+            f"factory function (that is, it takes no Tensors and constructs "
+            f"a new one), then please use CustomOp.impl_factory to register "
+            f"an implementation for it"
+        )
+    if key == "Meta":
+        raise NotImplementedError(
+            f"{custom_op}: when running with device='Meta' tensors: there is no "
+            f"abstract impl registered for this CustomOp. Please register one via "
+            f"CustomOp.impl_abstract to get this CustomOp to work with Meta tensors"
```

Comment:
This will give a nice error if a user tries to run a custom op with meta tensors, and the op doesn't have a meta impl. Out of curiosity though, will we get an equally nice error for fake tensors? E.g. if you're using the standard `torch.compile()` workflow on a model with a custom op, and forgot an abstract impl. If not, it would be nice to add that too (which is probably easier since it's all in python).

Reply:
That's correct, we don't have a good error message for fake tensors yet, it is on the list of follow-ups